### DISCUSSION TITLE
Traefik example not working due to missing serversTransport ability via labels

### DISCUSSION THREAD

**Discussion by Lem** [kanidm/kanidm#2381] [Q&A]
I tried to setup kanidm with traefik as shown in the integration example, but just getting internal server errors.

After some search I've came across [this issue](https://github.com/traefik/traefik/issues/8884), stating that a `serversTransports` via labels is not possible. If that is correct it would explain the errors I'm seeing and mean that [the example](https://github.com/kanidm/kanidm/blob/340d41482bd1a3dff486aa051335650c25a7073d/book/src/examples/traefik.md?plain=1#L55) is wrong.

I'm using traefik 2.10.x, the referenced issue itself was created for traefik 2.6, which is used in the provided example.

**Comment by yaleman** [comment]
Going by that issue the line should never have been there in the first place 🤔

**Comment by yaleman** [comment]
Does the example configuration work otherwise if you remove that line? These are community-provided so unless we can find someone who uses it, you're in the best place to test 😄

**Comment by Lem** [comment]
Not sure if it is possible to use traefik and services with self signed certificates just by defining labels in docker. All the examples I found yesterday where about importing the CA within a static config etc.

 My plan was to do some research on the weekend.

**Comment by Lem** [comment]
Ok, as I couldn't get back to sleep this setup is now working for me. I've used the comment from the referenced issue:

1) Use the a file provider by using `providers.file.directory` and `providers.file.watch`, either by appending them via the `commands`-section of docker compose file or by using the yaml syntax in a static config file (mounted into the traefik-container).

2) Inside the specified directory (needs to be mounted into the container) place a yaml with the already defined insecureSkipVerify :
```
http:
  serversTransports:
    kanidm:
      insecureSkipVerify: true
```

3) Define a label to use that yaml-config in the docker compose file:
```
labels:
[...]
- traefik.http.services.kanidm.loadBalancer.serversTransport=kanidm@file
```

Sadly it seems that you have to use a file provider for that flag (at least for me).

**Reply by TheRealGramdalf** [reply]
You can set it globally via `--serverstransport.insecureskipverify=true`, but that's not necessarily ideal since it sets it for everything behind traefik, not just kanidm

**Comment by yaleman** [comment]
You're really better off loading the CAs through [rootcas](https://doc.traefik.io/traefik/routing/services/#rootcas) than using `insecureSkipVerify`

**Comment by yaleman** [comment]
OK, I got it working - their rootCAs config ... just doesn't work for some reason?

But the files here do: https://github.com/yaleman/kanidm/tree/yaleman/issue2360/examples/traefik

**Comment by Firstyear** [comment]
Is this a bug in kani @yaleman ? Or is it a docs thing?

**Comment by yaleman** [comment]
It's a docs thing.
